### PR TITLE
theta model uses omega, not omega/p

### PR DESCRIPTION
**Pull request by mt5555**
pg2 bugfixes

1. theta model uses omega, not omega/p
2. bugfix when dynpes<npes GLL  grid dimension not set correctly on non-dyn PEs
3. add an abort if physgrid is used with interpolated output (not yet supported)

[nonBFB] for SMS_Ln5.ne4pg2_ne4pg2.FC5AV1C-L.cam-thetahy_sl_pg2 and all other theta tests

### CODE REVIEW

File: components/homme/src/share/gllfvremap_mod.F90 [hunk 1]
```diff
@@ -305,8 +305,11 @@ subroutine gfr_dyn_to_fv_phys_hybrid(hybrid, nt, hvcoord, elem, nets, nete, &
        call gfr_g2f_scalar(ie, elem(ie)%metdet, wr2, wr1)
        ! When omega_p is switched to omega, remove the division by
```
Comment (ambrad) [inline components/homme/src/share/gllfvremap_mod.F90]: You can remove this comment now, since this PR makes the switch.

**Comment by ambrad** [comment]
@mt5555 your last commit looks like it accidentally brought in erroneous submodule stuff.

**Comment by AaronDonahue** [comment]
ugg, I thought I had cleaned that up.  Thank you for catching that.  I will revert my commit, and recommit.

**Comment by mt5555** [comment]
fixed, thanks.

**Comment by oksanaguba** [comment]
Ran atm_developer on anvil for this PR + next against the same next. All as expected:

FAIL SMS_Ln5.ne4pg2_ne4pg2.FC5AV1C-L.anvil_intel.cam-thetahy_pg2 BASELINE master: DIFF
FAIL SMS_Ln5.ne4pg2_ne4pg2.FC5AV1C-L.anvil_intel.cam-thetahy_sl_pg2 BASELINE master: DIFF

**Comment by oksanaguba** [comment]
in next

**Pull request merged**